app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,278 @@
+class OrdersController < ApplicationController
+  
+  before_action :find_order, except: [:index, :order_confirmation, :order_summary]
+  
+  def index
+    if session[:merchant_id]
+      # for merchants' eyes only
+      # sending only the relevant order_items and orders 
+      @order_items = OrderItem.by_merchant(session[:merchant_id])
+      @orders = @order_items.map { |order_item| order_item.order }
+      @orders.uniq!
+      
+      
+      # if merchant wants to FILTER BY ORDER STATUS
+      @statuses = %w[ALL SHIPPED PAID PENDING]
+      if params[:status_selected]
+        statuses_index = params[:status_selected].to_i
+        
+        # @orders and @order_items will get filtered depending on status selected
+        @database_status = @statuses[statuses_index].downcase
+        if @database_status == "all"
+          # leave @order_items as is
+          return
+        else
+          @order_items = @order_items.find_all { |order_item| order_item.status == @database_status }
+          @orders = @order_items.map { |order_item| order_item.order }
+          @orders.uniq!
```

Comment:
Most of this is business logic that should live in models. As a rule of thumb, when I see more than a line or two that isn't related to control flow (render, redirect, etc.) or defining variables for views, I'm going to feel inclined to put the code in a model.

Reply:
Also almost all of these variable aren't used by the view and therefore shouldn't be instance variables.

Reply:
Also also, when refactoring this code to move most of it to the model, I would not just move it to one method in the model; I would figure out how to break it into several small methods.
In general, IMO, if a method is more than 30ish lines, it's worth considering how it could be broken down into smaller methods.